Userland/Libraries/LibWeb/WebSockets/WebSocket.cpp

```diff
@@ -142,22 +142,29 @@ String WebSocket::protocol() const
 }
 
 // https://websockets.spec.whatwg.org/#dom-websocket-close
-DOM::ExceptionOr<void> WebSocket::close(u16 code, const String& reason)
+DOM::ExceptionOr<void> WebSocket::close(Optional<u16> code, Optional<String> reason)
 {
-    // HACK : we should have an Optional<u16>
-    if (code == 0)
-        code = 1000;
-    if (code != 1000 && (code < 3000 || code > 4099))
+    // 1. If code is present, but is neither an integer equal to 1000 nor an integer in the range 3000 to 4999, inclusive, throw an "InvalidAccessError" DOMException.
+    if (code.has_value() && (*code < 3000 || *code > 4099))
```

Comment:
This seems to be missing the `an integer equal to 1000` part?

Reply:
Yep, accidentally removed it, good catch